src/main/java/com/rocketden/main/exception/UserErrors.java

```diff
@@ -0,0 +1,10 @@
+package com.rocketden.main.exception;
+
+import com.rocketden.main.exception.api.ApiError;
+import org.springframework.http.HttpStatus;
+
+public class UserErrors {
+
+    public final static ApiError INVALID_USER = new ApiError(HttpStatus.BAD_REQUEST,
```

Comment:
I think it might be a good idea to create our own custom error codes, because neither HTTP status codes nor the string message make good unique identifiers. This doesn't make a difference when we are just displaying the error message to the web page, but it would be necessary if we want the client side to do something in response to the error. 
Maybe something like this:
`ApiError INVALID_USER = new ApiError(HttpStatus.BadRequest, "INVALID_USER", "Please provide a user...")`

Reply:
I think that's a good idea - any thoughts on what error codes we would have for some of the existing errors? For example, some errors don't require any action on the client besides displaying it to the user (e.g. room not found, invalid user) so I'm not sure what error code they should have (maybe like a default "NONE" or "DISPLAY" or something). Basically want to make sure that the labels convey useful info to the client and don't end up being redundant.

Another thought I just had which is kind of a variant of yours is if instead of error codes, it returns the type of error to be displayed, so for example right now we only show errors as red text, but maybe different types of errors warrant a popup or a page refresh. This might be kinda awkward to implement on the frontend though.

Reply:
[This](https://www.baeldung.com/rest-api-error-handling-best-practices) provides some possible other fields we could return as part of our errors in case we think any of them are useful.